src/kaltura-player.js

```diff
@@ -50,17 +51,22 @@ export default class KalturaPlayer {
     setUIErrorOverlayConfig(this._uiManager, mediaInfo);
     return this._provider.getMediaConfig(mediaInfo)
       .then(mediaConfig => {
-        const dimensions = this._player.dimensions;
-        setUISeekbarConfig(this._uiManager, mediaConfig);
-        Utils.Object.mergeDeep(mediaConfig.metadata, this._player.config.metadata);
-        addKalturaPoster(mediaConfig.metadata, dimensions.width, dimensions.height);
-        addKalturaParams(mediaConfig.sources, this._player);
-        Utils.Object.mergeDeep(mediaConfig.plugins, this._player.config.plugins);
-        Utils.Object.mergeDeep(mediaConfig.session, this._player.config.session);
-        evaluatePluginsConfig(mediaConfig);
-        this._player.configure(mediaConfig);
+        this.setMedia(mediaConfig);
       }).catch(e => {
         this._player.dispatchEvent(new FakeEvent(this._player.Event.ERROR, new Error(Error.Severity.CRITICAL, Error.Category.PLAYER, Error.Code.LOAD_FAILED, e)));
       });
   }
+
+  setMedia(mediaConfig: Object): void {
```

Comment:
Change the mediaConfig type to its type (ProviderMediaConfigObject)

Reply:
done

Reply:
Once we actually separate the plugins outside playkit and put it here it will make much more sense.
Now we need to coerce the mediaConfig, which will be of type `ProviderMediaConfigObject`, to be playerConfig of type `PKPlayerOptionsObject`